lib/blockscore/errors/api_error.rb

```diff
@@ -15,19 +17,29 @@ class APIError < Error
     # APIError - Indicates an error on the server side (HTTP 5xx)
     # AuthenticationError - Indicates an authentication error (HTTP 401)
     def initialize(response)
-      body = JSON.parse(response.body, symbolize_names: true)
+      @http_body = JSON.parse(response.body, symbolize_names: true)
```

Comment:
I wonder if using `symbolize_names` is a good idea here. It can cause a memory leak since it dynamically creates `Symbol` objects which are never GC'd.

However, the problem with changing this is that the `http_body` attribute is exposed as part of the public API. If the keys suddenly changed to `String` objects it might break things for users. The main reason I bring this up now is for us to be aware of it in the future.

Technically it could be fixed by having `JSON.parse` return string keys, filter out all but the allowed keys given a known list. Then we (deep) symbolize the resulting Hash. This would be safe because it would keep the number of keys bounded to a known limit. I haven't looked deeply, maybe there's always only an `:errors` key, and it only has a `:message` and `:type` underneath it. I don't know if it's worth the trouble now though.